src/crystal/system/thread.cr

```diff
@@ -111,6 +115,7 @@ class Thread
     Thread.threads.push(self)
     Thread.current = self
     @main_fiber = fiber = Fiber.new(stack_address, self)
+    self.system_name = @name
```

Comment:
I don't think `#system_name=` needs to support `nil` at all, because we are never going to unset it. Instead this could be written as `self.system_name = name if name`